spinn_front_end_common/interface/interface_functions/local_tdma_builder.py

```diff
@@ -138,9 +138,24 @@ def __call__(
                     int(math.ceil(machine_time_step * time_scale_factor)))
 
                 # test timings
-                self._test_timings(
+                reduction = self._test_timings(
                     n_phases, time_between_phases, machine_time_step,
                     time_scale_factor, fraction_of_sending, app_vertex.label)
+                if reduction is not None:
+                    max_reduction = max(max_reduction, reduction)
+        if max_reduction != 0:
+            total_time_available = int(math.ceil(
+                machine_time_step * time_scale_factor * fraction_of_sending))
+            total_time_available += int(math.ceil(
+                machine_time_step * time_scale_factor * fraction_of_waiting))
+            new_time_scale_factor = int(math.floor(
+                (total_time_available * max_reduction) / machine_time_step))
+            if new_time_scale_factor != time_scale_factor:
+                logger.info(
+                    "Could reduce the cfg param "
+                    "fraction_of_time_spike_sending to {}, "
+                    "or time scale factor to {}".format(
```

Comment:
Should be "time_scale_factor" (underscores)

Reply:
fixed